site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+{% endaside %}
+
+The state stored in the `ApplicationContext` is the same data used in the `App` module to initialize the `WorkerProperties`, but it is now decoupled into an isolated module that helps to understand and maintain the application. In general, the `main` module of an application should be concerned with initializing application wide state. Also, as we mentioned earlier, the `App` class only needed to manage the `WorkerProperties` state so that it could coordinate this through from and to it's children.
+
+Now that we have the `ApplicationContext`, let's use it within our widgets. By creating an `Injector` and adding it to the `registry` created for the `projector`. `Injectors` are created using the `Injector` factory function from `@dojo/widget-core`.
+
+{% instruction 'Add the following imports to the `main` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:2-5 %}
+
+{% instruction 'Now, add the following after the `applicationContext` declaration.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:31-32 %}
+
+The first statement creates a `registry` where the application context can be registered. The second statement does two things. First, it uses the `Injector` factory function to wrap the `applicationContext` with the `BaseInjector`, a lightweight component that allows any object that extends `Evented` to be injected via a `Container`, which we will discuss in the next section. This statement also registers the `Injector` with the application's central registry. The registry provides a way to register a widget via a label, making it accessible to other parts of the application. You can learn more in the [registry tutorial](../comingsoon.html).
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:36 %}
+
+We need to pass the `registry` to the `projector` via the `setProperties` method to ensure that it is available for all widget and container instances.
+
+Now that the `Injector` is defined and registered and the `registry` has been set in the `projector`, it is time to create the components that will use it. In the next section, we will create a non-visual widget called a `Container` that will allow state to be injected into the `WorkerForm` and `WorkerContainer` widgets.
+
+{% section %}
+
+## Creating state containers
+
+{% task 'Create `Containers` that will allow state to be injected into widgets' %}
+
+On their own, `Injectors` are not able to help us very much because widgets expect state to be passed to them via properties. Therefore an `injector` must be connected to interested widgets in order for their state to be mapped to `properties` that widgets can consume, this is done using a `Container`. `Containers` are designed to coordinate the injection - they connect injectors to widgets and return `properties` from the `injectors` state which gets passed to the connected widgets.
```

Comment:
"Therefore an `injector` must be connected to interested widgets in order for their state to be mapped to `properties` that widgets can consume, this is done using a `Container`. `Containers` are designed to coordinate the injection - they connect injectors to widgets and return `properties` from the `injectors` state which gets passed to the connected widgets." -> "Therefore an `injector` must be connected to interested widgets in order for their state to be mapped to `properties` that widgets can consume by using a `Container`. `Containers` are designed to coordinate the injection - they connect injectors to widgets and return `properties` from the `injector`'s state which is passed to the connected widgets."